### PR TITLE
Use IOV_MAX and UIO_MAXIOV constants in limit vectored I/O

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**
Also updates the libc dependency to 0.2.77 (from 0.2.74) as the
constants were only recently added.

Related #68042, #75005

r? @Amanieu (also reviewed #75005)

### CODE REVIEW

**Comment by Thomasdezeeuw** [comment]
CI failed due to clock drift, could someone restart it?

**Comment by lzutao** [comment]
> CI failed due to clock drift, could someone restart it?

You have to update Cargo.lock to: `cargo update -p libc`.

**Comment by Thomasdezeeuw** [comment]
> > CI failed due to clock drift, could someone restart it?
> 
> You have to update Cargo.lock too: `cargo update -p libc`.

@lzutao I'm ok with doing that, but how is that related to the clock drift?

Also I couldn't get it to only update the libc dependency, is alright to update other dependencies as well?

**Comment by lzutao** [comment]
I don't know what clock drift is, but the main problem why CI failed is that
CI enforces `Cargo.lock` integrity.

**Comment by Thomasdezeeuw** [comment]
> I don't know what clock drift is, 

Clock drift is when the local clock's reported time differs from the "correct" time as reported by a known good clock.

> Running `cargo update -p libc` will update only libc crate.

This failed initially, but I got it to work. Thanks.

**Comment by Thomasdezeeuw** [comment]
CI is complaining about an unneeded unsafe block in `library/std/src/sys/unix/process/process_unix.rs`, but I didn't touch that file.

**Comment by Amanieu** [comment]
It's because you updated the libc dependency. You should fix these errors.

File: library/std/src/sys/unix/fd.rs [hunk 1]
```diff
@@ -7,7 +7,6 @@ use crate::cmp;
 use crate::io::{self, Initializer, IoSlice, IoSliceMut, Read};
 use crate::mem;
 #[cfg(not(any(target_os = "redox", target_env = "newlib")))]
```
Comment (lzutao) [inline library/std/src/sys/unix/fd.rs:9]: The cfg may be unwanted

**Comment by Thomasdezeeuw** [comment]
On macOS those function do require unsafe (that's why it didn't show up in my local testing). The definition on Linux: https://docs.rs/libc/0.2.77/libc/fn.WTERMSIG.html (no `unsafe`), vs https://docs.rs/libc/0.2.77/x86_64-apple-darwin/libc/fn.WTERMSIG.html (`unsafe`). I've opened https://github.com/rust-lang/libc/issues/1888 for this.

@Amanieu would you be ok with putting `#[allow(unused_unsafe)]` around them for now? With a comment linking to https://github.com/rust-lang/libc/issues/1888.

File: library/std/src/sys/unix/fd.rs (hunk 1 above)
Comment (Thomasdezeeuw) [reply library/std/src/sys/unix/fd.rs:9]: Good catch, will fix.

**Comment by Amanieu** [comment]
Put it under a linux-only `#[cfg_attr]` and add a comment linking to the issue.

**Comment by Thomasdezeeuw** [comment]
@Amanieu CI is now green.

**Pull request merged**